### PR TITLE
[DO NOT MERGE TILL VARs and Secrets are created] Add kernel header tests to ci/cd

### PR DESCRIPTION
**Pull request by samiura**
1.  We are testing kernel-collector against 12 different flavors of Linux distribution.
2.  Kernel header tests are only triggered if and only if kernel-collector directory or underlying github action ci/cd. yaml  changes. 
3. We are also building and pushing reducer and kernel-collector images to docker hub repo.

### CODE REVIEW

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -214,3 +237,30 @@ jobs:
...
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - uses: dorny/paths-filter@v2
+      id: changes
+      with:
+        filters: |
+          kernel:
+            - 'collector/kernel/**'
```
Comment (jimwsplk) [inline .github/workflows/build-and-test.yaml:258]: Is this filter to determine when to run the kernel-header tests when kernel-collector code changes?  If so there is other source code included in the kernel-collector in addition to opentelemetry-ebpf/collector/kernel/*.

File: test/kernel-headers/run-test.sh [hunk 2]
```diff
@@ -8,27 +8,42 @@ export EBPF_NET_SRC_ROOT="${EBPF_NET_SRC_ROOT:-$(git rev-parse --show-toplevel)}
...
 [ "$kernel_version" != "" ] && name=${name}-${kernel_version}
 print "Testing ${name}"
 ${EBPF_NET_SRC_ROOT}/test/kernel-headers/bootstrap.sh ${distro} ${version} ${kernel_version}
 
 cd ${name}
 print "running 0-setup.sh"
-./0-setup.sh
+./0-setup.sh ${tag}
+
+## this script should only be run for SE linux. Currently only centos-7 is an SE.
```
Comment (jimwsplk) [inline test/kernel-headers/run-test.sh:37]: 2-apply-selinux-policy.sh was intended to be run on all distros, and should be safe to run whether the distro is SELinux or not.  Was it causing problems?

File: test/kernel-headers/source/runners/3-fetch.sh [hunk 3]
```diff
@@ -8,4 +8,12 @@ set -x
 
 vagrant ssh -- -- ./uninstall-kernel-headers.sh
 vagrant ssh -- -- sudo rm -rf /var/cache/ebpf_net/kernel-headers || true
-vagrant ssh -- -R "5000:localhost:5000" -- ./agent.sh
+
+if [ $# -eq 2 ]
```
Comment (jimwsplk) [inline test/kernel-headers/source/runners/3-fetch.sh:12]: Are all of these ifs of # args necessary - can't you pass all args with "$@" ?

File: dev/selinux-resolv-conf.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+if [ "$EUID" -ne 0 ]; then
+   echo "This script must be run as root"
+   exit 1
+fi
+
+
+
+##CentOS 7 uses NetworkManager for the network, the ones that dont have static (manually) configured network.
```
Comment (jimwsplk) [inline dev/selinux-resolv-conf.sh:12]: Is this CentOS or SELinux related?  If the former, change script name and decouple from the the other selinux stuff?

File: .github/workflows/build-and-test.yaml (hunk 1 above)
Comment (samiura) [reply .github/workflows/build-and-test.yaml:258]: This is to identify the files where kernel collector related codes are changed. What other directories I need to include here?

File: test/kernel-headers/run-test.sh (hunk 2 above)
Comment (samiura) [reply test/kernel-headers/run-test.sh:37]: It was throwing some errors on few of the occasions for Ubuntu-jammy. Do you recommend this to revert?

File: .github/workflows/build-and-test.yaml (hunk 1 above)
Comment (samiura) [reply .github/workflows/build-and-test.yaml:258]: :done

File: test/kernel-headers/run-test.sh (hunk 2 above)
Comment (samiura) [reply test/kernel-headers/run-test.sh:37]: done

File: test/kernel-headers/source/runners/3-fetch.sh (hunk 3 above)
Comment (samiura) [reply test/kernel-headers/source/runners/3-fetch.sh:12]: removed.

File: dev/selinux-resolv-conf.sh (hunk 4 above)
Comment (samiura) [reply dev/selinux-resolv-conf.sh:12]: done.

File: test/kernel-headers/run-test.sh [hunk 5]
```diff
@@ -34,16 +34,20 @@ cd ${name}
 print "running 0-setup.sh"
 ./0-setup.sh ${tag}
 
-## this script should only be run for SE linux. Currently only centos-7 is an SE. 
+## This should only be run for SE linux and other distrobutions using NetworkManager instead of static network configuration.
```
Comment (jimwsplk) [inline test/kernel-headers/run-test.sh:37]: "SE linux and other distrobutions" --> "CentOS and other distributions"?

**Review by jimwsplk — APPROVED** [review]

**Review by jimwsplk — APPROVED** [review]

**Review by jimwsplk — APPROVED** [review]

**Pull request merged**